distribution/rules.bzl

```diff
@@ -86,57 +78,152 @@ def _distribution_impl(ctx):
-load("@bazel_tools//tools/build_defs/pkg:pkg.bzl", "pkg_tar", "pkg_deb")
 def deploy_deb(name,
-               package_dir,
-               maintainer,
-               version_file,
-               description,
-               postinst = None,
-               prerm = None,
-               target = None,
-               empty_dirs = [],
-               files = {},
-               depends = [],
-               symlinks = {},
-               modes = {}):
+                package_name,
+                installation_dir,
+                maintainer,
+                version_file,
+                description,
+                target = None,
+                empty_dirs = [],
+                files = {},
+                depends = [],
+                symlinks = {},
+                permissions = {}):
     java_deps_tar = []
     if target:
+        java_deps_name = "_{}-deb-deps".format(package_name)
         java_deps(
-            name = "_{}-deps".format(name),
-            targets = [target],
+            name = java_deps_name,
+            target = target,
             java_deps_root = "services/lib/"
         )
-        java_deps_tar.append("_{}-deps".format(name))
+        java_deps_tar.append(java_deps_name)
+
+    tar_name = "_{}-deb-tar".format(package_name)
 
     pkg_tar(
-        name = "_{}-tar".format(name),
+        name = tar_name,
         extension = "tgz",
         deps = java_deps_tar,
-        package_dir = package_dir,
+        package_dir = installation_dir,
         empty_dirs = empty_dirs,
         files = files,
+        mode = "0755",
         symlinks = symlinks,
-        modes = modes,
+        modes = permissions,
     )
 
     pkg_deb(
-        name = "deploy-deb",
-        data = "_{}-tar".format(name),
-        package = name,
+        name = name,
+        data = tar_name,
+        package = package_name,
         depends = depends,
         maintainer = maintainer,
         version_file = version_file,
-        postinst = postinst,
-        prerm = prerm,
         description = description
     )
 
 
+def deploy_rpm(name,
+                package_name,
+                installation_dir,
+                version_file,
+                spec_file,
+                target = None,
+                empty_dirs = [],
+                files = {},
+                permissions = {},
+                symlinks = {}
+                ):
+    java_deps_tar = []
+    if target:
+        java_deps_name = "_{}-rpm-deps".format(package_name)
+        java_deps(
+            name = java_deps_name,
+            target = target,
+            java_deps_root = "services/lib/"
+        )
+        java_deps_tar.append(java_deps_name)
+
+    tar_name = "_{}-rpm-tar".format(package_name)
+    rpm_version_file = "_{}-rpm-version".format(package_name)
+
+    native.genrule(
+        name = rpm_version_file,
+        srcs = [version_file],
+        outs = [rpm_version_file.replace("-version", ".VERSION")],
+        cmd = "sed -e 's|-|_|g' $< > $@"
```

Comment:
let's add an explanation with an inline comment next to the `sed` command:
```
# replaces any `-` with `_` since RPM does not allow a version number containing a `-` such as `1.5.0-SNAPSHOT`